File: app_torneig_flutter/lib/models/equip_dto.dart
import 'package:app_torneig/models/jugador_simple_dto.dart';
import 'package:app_torneig/models/partit_dto.dart';

class EquipDTO {
  int? idEquip;
  String? nom;
  String? curs;
  int? punts;
  String? imatge;
  bool? esGuanyador;
  int? puntsContra;
  int? idTemporada;
  String? nomTemporada;
  int? idGrup;
  String? nomGrup;
  int? partitsJugats;
  int? partitsGuanyats;
  int? partitsEmpatats;
  int? partitsPerduts;
  List<JugadorSimpleDTO>? elsJugadors;
  List<PartitDTO>? elsPartits;

  EquipDTO({
    required this.idEquip,
    this.nom,
    this.curs,
    this.punts,
    this.imatge,
    this.esGuanyador,
    this.puntsContra,
    this.idTemporada,
    this.nomTemporada,
    this.idGrup,
    this.nomGrup,
    this.partitsJugats,
    this.partitsGuanyats,
    this.partitsEmpatats,
    this.partitsPerduts,
    this.elsJugadors,
    this.elsPartits,
  });

  EquipDTO.fromJSON(Map<String, dynamic> objecteJSON) {
    idEquip = objecteJSON['idEquip'];
    nom = objecteJSON['nom'] ?? "";
    curs = objecteJSON['curs'] ?? "";
    imatge = objecteJSON['imatge'] ?? "assets/images/defecte.png";
    esGuanyador = objecteJSON['esGuanyador'];
    punts = objecteJSON['punts'];
    puntsContra = objecteJSON['puntsContra'];
    idTemporada = objecteJSON['idTemporada'];
    nomTemporada = objecteJSON['nomTemporada'] ?? "";
    idGrup = objecteJSON['idGrup'];
    nomGrup = objecteJSON['nomGrup'] ?? "";
    partitsJugats = objecteJSON['partitsJugats'];
    partitsGuanyats = objecteJSON['partitsGuanyats'];
    partitsEmpatats = objecteJSON['partitsEmpatats'];
    partitsPerduts = objecteJSON['partitsPerduts'];
    elsJugadors = objecteJSON['elsJugadors'] != null
        ? List<JugadorSimpleDTO>.from(objecteJSON['elsJugadors']
            .map((jugador) => JugadorSimpleDTO.fromJSON(jugador)))
        : null;
    elsPartits = objecteJSON['elsPartits'] != null
        ? List<PartitDTO>.from(objecteJSON['elsPartits']
            .map((partit) => PartitDTO.fromJSON(partit)))
        : null;
  }
}

File: Torneig_Mort_Spring/src/main/java/org/eljust/DTO/EquipDTO.java
package org.eljust.DTO;

import java.util.ArrayList;
import java.util.List;

import lombok.Getter;
import lombok.Setter;

/**
 * Equip DTO amb tota la informació rellevant. S'utilitza per a quan volem saber
 * la informació d'un equip en concret.
 */
@Getter
@Setter
public class EquipDTO {
	private Long idEquip;
	private String nom;
	private String curs;
	private String imatge;
	private boolean esGuanyador;
	private int punts;
	private int puntsContra;
	private String nomTemporada;
	private Long idTemporada;
	private Long idGrup;
	private String nomGrup;
	private int partitsJugats;
	private int partitsGuanyats;
	private int partitsPerduts;
	private int partitsEmpatats;
	private List<JugadorSimpleDTO> elsJugadors = new ArrayList<>();
	private List<PartitDTO> elsPartits = new ArrayList<>();

}
